sound/soc/sof/sof-audio.h

```diff
@@ -392,6 +392,13 @@ struct snd_sof_widget {
 	int num_sink_pins;
 	int num_source_pins;
 
+	/* The sink/source pin binding array, it takes the form of
+	 * [widget_name_connected_to_pin0, widget_name_connected_to_pin1, ...],
+	 * with the index as the queue ID.
```

Comment:
You have to explain what happens when a pin is not connected, and rules such as if one pin uses a special binding all pins use the special binding. You have a turn-key design document already, use it to document your code and explain what the code does.